hazelcast/include/hazelcast/client/connection/AuthenticationFuture.h

```diff
@@ -30,37 +28,34 @@
 
 namespace hazelcast {
     namespace client {
-        class Address;
-
-        namespace spi {
-            class ClientContext;
+        namespace exception {
+            class IException;
         }
-
         namespace connection {
             class Connection;
 
-            class HAZELCAST_API OwnerConnectionFuture {
+            class HAZELCAST_API AuthenticationFuture {
             public:
-                OwnerConnectionFuture(spi::ClientContext&);
-
-                void markAsClosed();
+                AuthenticationFuture();
 
-                boost::shared_ptr<Connection> getOrWaitForCreation();
+                void onSuccess(const boost::shared_ptr<Connection> &connection);
 
-                boost::shared_ptr<Connection> createNew(const Address& address);
+                void onFailure(const boost::shared_ptr<exception::IException> &throwable);
 
-                void closeIfAddressMatches(const Address& address);
+                void onFailure(const exception::IException &throwable);
```

Comment:
It looks like this has no usages. Can this be deleted ?

Reply:
Removed it.